### PR TITLE
feat(icons): added `dam` icon

### LINKED ISSUE #910: Dam Icon
## Icon Request

* Icon name: Dam
* Use case: For energy websites
* _Screenshots_ of similar icons:
image
image
image

### PR DESCRIPTION
**Pull request by AndreasSas**
closes #910

## What is the purpose of this pull request?
- [x] New Icon
- [ ] Bug fix
- [ ] New Feature
- [ ] Documentation update
- [ ] Other:

### Description
An icon of a Water Dam

### Icon use case
The Icon can be used for energy websites, or for naturally made dams by wildlife

### Concept 
- [x] I have provided valid use cases for each icon.
- [x] I have not added any a brand or logo icon.
- [x] I have not used any hate symbols.
- [x] I have not included any religious or political imagery.

### Author, credits & license
- [ ] The icons are solely my own creation.
- [x] The icons were originally created in #910 by @karsa-mistmere , @raphasampaio , @AndreasSas (final edit)
- [x] I've based them on the following Lucide icons: `water`
- [ ] I've based them on the following design:

### Naming 
- [x] I've read and followed the [naming conventions](https://lucide.dev/guide/design/icon-design-guide#naming-conventions)
- [x] I've named icons by what they are rather than their use case.
- [x] I've provided meta JSON files in `icons/[iconName].json`.

### Design 
- [x] I've read and followed the [icon design guidelines](https://lucide.dev/guide/design/icon-design-guide)
- [x] I've made sure that the icons look sharp on low DPI displays.
- [x] I've made sure that the icons look consistent with the icon set in size, optical volume and density.
- [x] I've made sure that the icons are visually centered.
- [x] I've correctly optimized all icons to three points of precision.

## Before Submitting 
- [x] I've read the [Contribution Guidelines](https://github.com/lucide-icons/lucide/blob/main/CONTRIBUTING.md).
- [x] I've checked if there was an existing PR that solves the same issue.

### CODE REVIEW

File: icons/dam.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+  "$schema": "../icon.schema.json",
+  "contributors": [
+    "AndreasSas"
+  ],
+  "tags": [
+    "electricity energy water"
```
Comment (jguddas) [inline icons/dam.json:7]:
```suggestion
    "electricity",
    "energy",
    "water"
```

File: icons/dam.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  ],
+  "categories": [
+    "buildings sustainability"
```
Comment (jguddas) [inline icons/dam.json:10]:
```suggestion
    "buildings",
    "sustainability"
```

File: icons/dam.svg [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  stroke-linecap="round"
+  stroke-linejoin="round"
+>
+  <path d="M11 11.31c1.17.56 1.54 1.69 3.5 1.69 2.5 0 2.5-2 5-2 1.3 0 1.9.5 2.5 1" />
+  <path d="M12 18c.6.5 1.2 1 2.5 1 2.5 0 2.5-2 5-2 1.3 0 1.9.5 2.5 1" />
+  <path d="M2 10h4" />
+  <path d="M2 14h4" />
+  <path d="M2 18h4" />
+  <path d="M2 6h4" />
+  <path d="M7 3c-.55 0-1 .45-1 1v16c0 .55.45 1 1 1h4c.55 0 1-.45 1-1L10 4c0-.55-.45-1-1-1z" />
```
Comment (jguddas) [inline icons/dam.svg:18]:
```suggestion
  <path d="M11 11.31c1.17.56 1.54 1.69 3.5 1.69 2.5 0 2.5-2 5-2 1.3 0 1.9.5 2.5 1" />
  <path d="M11.75 18c.35.5 1.45 1 2.75 1 2.5 0 2.5-2 5-2 1.3 0 1.9.5 2.5 1" />
  <path d="M2 10h4" />
  <path d="M2 14h4" />
  <path d="M2 18h4" />
  <path d="M2 6h4" />
  <path d="M7 3a1 1 0 0 0-1 1v16a1 1 0 0 0 1 1h4a1 1 0 0 0 1-1L10 4a1 1 0 0 0-1-1z" />
```

**Comment by jguddas** [comment]
The bottom wave connects a bit weirdly IMO.

**Comment by AndreasSas** [comment]
Sorry about the tags and categories not being stored correctly in the json already. To be honest, the "contribute to lucide" modal on the studio site didn't specify if I the inserted text would already seperate itself. I, mistakenly, thought "oh it'll just split on the space automatically"

**Comment by AndreasSas** [comment]
> The bottom wave connects a bit weirdly IMO. 

Hey I'm a bit confused, is the thumbs up from Karsa an affirmation that the icon is good as it is, or that they agree with it connecting weirdly?

**Comment by ericfennis** [comment]
Hmm, I've experimented by moving the dam a bit to the left (bottom design). But that doesn't look better IMO. I think we should go with @AndreasSas design.
image

**Review by ericfennis — APPROVED** [review]

**Review by jguddas — APPROVED** [review]

**Pull request merged**